File: frontend/src/api/services/expenseService.js
import api from "../api";

const expenseService = {
  getGroupExpenses: (groupId) =>
    api.get(`groups/${groupId}/expenses`).then((res) => res.data),
  calculateSettlements: (groupId) =>
    api.get(`/groups/${groupId}/settlements/calculate`).then((res) => res.data),
  createExpense: (groupId, expense) =>
    api.post(`/groups/${groupId}/expenses`, expense).then((res) => res.data),
};

export default expenseService;

File: backend/src/main/java/io/plan/mate/expense/tracker/backend/expense/service/ExpenseService.java
package io.plan.mate.expense.tracker.backend.expense.service;

import io.plan.mate.expense.tracker.backend.commons.service.dto.PagedResponse;
import io.plan.mate.expense.tracker.backend.expense.service.dto.ExpenseDto;
import io.plan.mate.expense.tracker.backend.expense.controller.payload.request.CreateExpenseRequest;
import io.plan.mate.expense.tracker.backend.expense.controller.payload.request.UpdateExpenseRequest;
import org.springframework.data.domain.Pageable;

public interface ExpenseService {

  ExpenseDto createExpense(Long groupId, CreateExpenseRequest createExpenseRequest);

  ExpenseDto updateExpense(
      Long groupId, Long expenseId, UpdateExpenseRequest updateExpenseRequest);

  void deleteExpense(Long groupId, Long expenseId);

  PagedResponse<ExpenseDto> getGroupExpenses(Long groupId, String search, Pageable pageable);
}
